gammapy/astro/darkmatter/spectra.py

```diff
@@ -66,27 +73,98 @@ class PrimaryFlux(TemplateNDSpectralModel):
         "V->e": "V->e",
         "V->mu": r"V->\[Mu]",
         "V->tau": r"V->\[Tau]",
+        "aZ": "aZ",
+        "HZ": "HZ",
+        "d": "d",
+        "u": "u",
+        "s": "s"
     }
 
-    table_filename = "$GAMMAPY_DATA/dark_matter_spectra/AtProduction_gammas.dat"
-
     tag = ["PrimaryFlux", "dm-pf"]
 
-    def __init__(self, mDM, channel):
-        self.table_path = make_path(self.table_filename)
+    def __init__(self, mDM, channel, source='pppc4'):
+
+        if source is None:
+            source='pppc4'
+
+        if source.lower() == "pppc4" or not source:
+            table_filename = "$GAMMAPY_DATA/dark_matter_spectra/AtProduction_gammas.dat"
+            if channel in ('aZ','HZ'):
+                raise ValueError(
+                    f"\n\nThe channel "+channel+" is not available in PPPC4, please choose another channel or use CosmiXs (cosmixs) as source\n"
+                )
+            elif channel in ('d','u', 's'):
+                raise ValueError(
+                    f"\n\nThe channel "+channel+" is not available in PPPC4, please choose the equivalent channel q or use CosmiXs (cosmixs) as source\n"
+                )
+
+        elif source == 'cosmixs':
+            table_filename = 'gammapy/external/CosmiXs/Data/AtProduction-Gamma.dat'
+
+            if channel in ("V->e", "V->mu", "V->tau"):
+                raise ValueError(
+                    f"\n\nThe channel "+channel+" is not available in CosmiXs, please choose another channel or use PPPC4 as source\n"
+                )
+            elif channel =='q':
+                raise ValueError(
+                    f"\n\nThe channel q is not available in cosmixs, please choose an equivalent channel such as d, u or s or use PPPC4 as source\n"
+                )
```

Comment:
Maybe it's worth just printing the available particles of either model in the error. 
This requires splitting the `channel_registry` into to dicts.